### PR TITLE
Kyv navand 918 fix voice instructions

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
Addressing same fix https://github.com/mapbox/mapbox-navigation-android/pull/6684 but for `VoiceInstructions`

thanks to @VysotskiVadim

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt [hunk 1]
```diff
@@ -701,14 +707,18 @@ internal class MapboxTripSession(
...
+     * call [lastVoiceInstruction]
      */
     private fun invalidateLatestInstructions(
-        latestInstructionWrapper: BannerInstructionEvent.LatestInstructionWrapper?
+        latestInstructionWrapper: BannerInstructionEvent.LatestInstructionWrapper?,
+        voiceInstruction: VoiceInstructions?,
     ) {
         bannerInstructionEvent.invalidateLatestBannerInstructions(latestInstructionWrapper)
-        lastVoiceInstruction = null
+        if (lastVoiceInstruction == voiceInstruction) {
```
Comment (dzinad) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:719]: I think we now have a case when the index won't be set to null but the voice instruction will.
The index will be null if `latestInstructionWrapper` is equal to saved one. But `latestInstructionWrapper` is about banner instruction and index. So can there be the following case:
1) banner instruction is updated;
2) index is not updated;
3) voice instruction is not updated;
4) we invoke `invalidateLatestInstructions`;
5) result: banner instruction is not null, voice instruction is null?

I'm not sure, maybe this just can't happen. just double-checking.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt [hunk 2]
```diff
@@ -701,14 +707,18 @@ internal class MapboxTripSession(
...
-     * Invalidate latest banner instruction. To get the latest banner instruction wrapper call
-     * [BannerInstructionEvent.latestInstructionWrapper]
+     * Invalidate latest banner and voice instruction. To get the latest banner instruction wrapper call
+     * [BannerInstructionEvent.latestInstructionWrapper], to get the latest voice instruction
+     * call [lastVoiceInstruction]
      */
     private fun invalidateLatestInstructions(
-        latestInstructionWrapper: BannerInstructionEvent.LatestInstructionWrapper?
+        latestInstructionWrapper: BannerInstructionEvent.LatestInstructionWrapper?,
+        voiceInstruction: VoiceInstructions?,
```
Comment (dzinad) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:716]: Is it right for the index to be wrapped in banner instruction event if it's responsible both for banner and voice instruction?
I'd keep them all together.

Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:716]: The index is taken from `BannerInstruction#getIndex` and later the SDK re-uses that `index` value for voice instruction as well. Not sure why it's done so. (the index of voice and banner instruction must be the same, but why it's taken from `BannerInstruction#getIndex` is not clear)
@LukasPaczos do you have the context?
We can address the issue separately

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt (hunk 1 above)
Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:719]: that actually a good point. It seems not critical for the voice instruction as it's pronounced once. 
Let's cut a ticket to address the issue separately. It looks like refactoring task, along with https://github.com/mapbox/mapbox-navigation-android/pull/6689/files#r1041952357

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt (hunk 2 above)
Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:716]: cut `NAVAND-964`

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 #### Bug fixes and improvements
 - Fixed `BannerInstructions` issue where the banner instruction might have been removed from `RouteProgress` at some point around a edge's leg. [#6684](https://github.com/mapbox/mapbox-navigation-android/pull/6684)
+- Fixed `VoiceInstructions` issue where the voice instruction might have been removed from `RouteProgress` at some point around a edge's leg. [#6689](https://github.com/mapbox/mapbox-navigation-android/pull/6689)
```
Comment (LukasPaczos) [inline CHANGELOG.md:9]:
```suggestion
- Fixed an issue where `RouteProgress#BannerInstructions` could've become `null` when `MapboxNavigation#updateLegIndex` was called. [#6684](https://github.com/mapbox/mapbox-navigation-android/pull/6684)
- Fixed an issue where `RouteProgress#VoiceInstructions` could've become `null` when `MapboxNavigation#updateLegIndex` was called. [#6689](https://github.com/mapbox/mapbox-navigation-android/pull/6689)
```

**Review by dzinad — APPROVED** [review]

**Pull request merged**